### PR TITLE
ISPN-9357 404 returned from REST for caches with object storage and non…

### PR DESCRIPTION
**Pull request by gustavonalle**
…-string keys

https://issues.jboss.org/browse/ISPN-9357

### CODE REVIEW

File: server/integration/testsuite/src/test/java/org/infinispan/server/test/dataconversion/Currency.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.io.Serializable;
+
+import org.infinispan.commons.marshall.SerializeWith;
+
+/**
+ * @since 9.4
+ */
+@SerializeWith(value = Currency.Externalizer.class)
```
Comment (anistor) [inline server/integration/testsuite/src/test/java/org/infinispan/server/test/dataconversion/Currency.java]: SerializeWith does not exist. According to documentation :)))

Comment (gustavonalle) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/dataconversion/Currency.java]: This is not actually needed, I could just use ```implements Serializable```, but adding this allowed me to test  deploying entities that depends on a internal infinispan module

Comment (anistor) [reply server/integration/testsuite/src/test/java/org/infinispan/server/test/dataconversion/Currency.java]: I was only joking. We can use that. It even works.

**Comment by gustavonalle** [comment]
Included in #6137

**Pull request closed without merging**